mypy/semanal.py

```diff
@@ -905,8 +905,20 @@ def analyze_metaclass(self, defn: ClassDef) -> None:
                 self.fail("Dynamic metaclass not supported for '%s'" % defn.name, defn)
                 return
             sym = self.lookup_qualified(defn.metaclass, defn)
-            if sym is not None and not isinstance(sym.node, TypeInfo):
+            if sym is None:
+                # Probably a name error - it is already handled elsewhere
+                return
+            if not isinstance(sym.node, TypeInfo):
                 self.fail("Invalid metaclass '%s'" % defn.metaclass, defn)
+                return
+            inst = fill_typevars(sym.node)
+            assert isinstance(inst, Instance)
```

Comment:
I think it should be an error then. Do you agree? I don't want to go silently to a fallback,